exp/ingest/processors.go

```diff
@@ -0,0 +1,88 @@
+package ingest
+
+import (
+	"github.com/stellar/go/exp/ingest/io"
+	"github.com/stellar/go/xdr"
+)
+
+type ChangeProcessor interface {
+	Init(sequence uint32) error
```

Comment:
I wonder if we should change it to:
```suggestion
	Init(header xdr.LedgerHeader) error
```
With different `Init` definition than in `LedgerTransactionProcessor` it won't be possible to create a processor that processes both changes and transactions (you cannot redeclare method in Golang). But what's more important `LedgerHeader` contains information that can be useful for processing in the future (ex. protocol version, close time, etc.).

Reply:
how do you get a `xdr.LedgerHeader` from the history archive snapshot?

Reply:
It's there in `ledgers` directory (see: https://github.com/stellar/stellar-core/blob/master/docs/history.md). For now, we can get it from stellar-core DB (because we get it anyway to compare bucket list hash).